scripts/start.sh

```
@@ -9,13 +9,50 @@ DEPLOY_LOG="$PROJECT_ROOT/deploy.log"
 
 TIME_NOW=$(date +%c)
 
-# build 파일 복사
+echo "--- 배포 시작: $TIME_NOW ---" >> $DEPLOY_LOG
+
+# 1. Java 설치 확인 및 자동 설치
+if ! command -v java &> /dev/null
+then
+    echo "$TIME_NOW > Java가 설치되어 있지 않습니다. OpenJDK 17 설치를 시작합니다." >> $DEPLOY_LOG
+    sudo apt-get update -y
+    sudo apt-get install openjdk-17-jdk -y >> $DEPLOY_LOG 2>&1
```

Shellcheck (0.11.0)

[warning] sudo doesn't affect redirects. Use .. | sudo tee -a file

(SC2024)